app/pages/api/supply.ts

```diff
@@ -0,0 +1,111 @@
+import { getContract, getStaticProvider } from "@klimadao/lib/utils";
+import { formatUnits } from "ethers/lib/utils";
+import { DEFAULT_NETWORK } from "lib/constants";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+export const config = {
+  api: {},
+  // Specifies the maximum allowed duration for this function to execute (in seconds)
+  maxDuration: 5,
+};
+
+type APIResponse =
+  | number
+  | {
+      totalSupply: number;
+      circulatingSupply: number;
+    }
+  | {
+      error: string;
+    };
+
+// This can be simplified to be just a string[]
+// but IERC20 ABI doesn't have a decimal exposed
```

Comment:
really? weird... the ERC-20 spec includes the decimal function so this should be readable from the contract, maybe we need a more complete IERC20 ABI

for now you can just use this util to get the decimals: https://github.com/KlimaDAO/klimadao/blob/staging/lib/utils/getTokenDecimals/index.ts